src/askem_beaker/contexts/mira_model/procedures/python3/amr_to_templates.py

```diff
@@ -0,0 +1,47 @@
+from mira.sources.amr.petrinet import template_model_from_amr_json
+from mira.modeling.amr.petrinet import AMRPetriNetModel
+
+
+model_json = {{ var_name|default("model") }}
+# model_tm = template_model_from_amr_json(model_json)
+model_tm = model_json
+
+# Define a new MIRA TemplateModel for each template in the model
+templates_tm = []
+templates_json = []
+for i, __ in enumerate(model_tm.templates):
+    t = copy.deepcopy(model_tm.templates[i])
+    tm = TemplateModel(
+        templates = [t],
+        # parameters = {params_global[p]: params_concepts_global[params_global[p]] for p in t.get_parameter_names()},
+        # initials = {vars_global[c.name]: inits_concepts_global[c.name] for c in t.get_concepts()},
+        # annotations = Annotations(name = f"{t.name}{i}")
+        parameters = {p: model_tm.parameters[p] for p in t.get_parameter_names()},
+        initials = {v: model_tm.initials[v] for v in t.get_concept_names()},
+        annotations = Annotations(name = f"{t.name}"),
```

Comment:
Why do not need the index `i` here, compare to the commented out line?

Reply:
This is @liunelson code, will let Nelson comment.